File: Programming Skills/Find the Difference/findTheDifference.swift
class Solution {
    func findTheDifference(_ s: String, _ t: String) -> Character {
        var counts: [Character: Int] = [:]

        for char in s {
            counts[char, default: 0] += 1
        }

        for char in t {
            counts[char, default: 0] -= 1

            if let count = counts[char], count < 0 { return char }
        }

        return Character("")
    }
}

File: Programming Skills/Find the Difference/findTheDifference.c
char findTheDifference(char* s, char* t) {
    int counts[26] = {0};
    int answer = 0;

    for (int i = 0; s[i] != '\0'; i++) {
        counts[s[i] - 'a']++;
    }
    
    for (int j = 0; t[j] != '\0'; j++) {
        counts[t[j] - 'a']--;

        if (counts[t[j] - 'a'] < 0) {
            answer = t[j];
            break;
        }
    }

    return answer;
}